docs/aws/audit/route53monitoring/rules/domain_expired_remediation.mdx

```diff
@@ -1,6 +1,240 @@
+
+3. **Check Expiration Dates:**
+   - For each domain listed, check the "Expiration date" column to identify any domains that are nearing expiration or have already expired.
+
+4. **Set Up Notifications:**
+   - To ensure you are notified before a domain expires, set up email notifications.
+   - Click on the domain name to view its details.
+   - Under the "Expiration" section, ensure that the contact information is up-to-date so that renewal reminders are sent to the correct email addresses.
+
+By regularly reviewing the expiration dates and setting up notifications, you can prevent domains from expiring unexpectedly.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent expired domains in Route 53 using AWS CLI, you can follow these steps:
+
+1. **List All Hosted Zones:**
+   Use the following command to list all hosted zones in your AWS account. This will help you identify the domains you need to monitor for expiration.
+
+   ```sh
+   aws route53 list-hosted-zones
+   ```
+
+2. **Get Domain Expiration Details:**
+   For each hosted zone, retrieve the domain registration details to check the expiration date. Replace `<domain-name>` with your actual domain name.
+
+   ```sh
+   aws route53domains get-domain-detail --domain-name <domain-name>
+   ```
+
+3. **Set Up CloudWatch Alarms:**
+   Create CloudWatch alarms to monitor the expiration dates of your domains. This can be done by creating a custom metric that tracks the expiration date and setting up an alarm to notify you before the domain expires.
+
+   ```sh
+   aws cloudwatch put-metric-alarm --alarm-name "DomainExpirationAlarm" --metric-name "DomainExpiration" --namespace "AWS/Route53" --statistic "Minimum" --period 86400 --threshold 30 --comparison-operator "LessThanThreshold" --evaluation-periods 1 --alarm-actions <SNS-topic-ARN>
+   ```
+
+4. **Automate Notifications:**
+   Set up an SNS topic to receive notifications when the CloudWatch alarm is triggered. This ensures you are alerted well in advance of any domain expiration.
+
+   ```sh
+   aws sns create-topic --name DomainExpirationNotifications
+   aws sns subscribe --topic-arn <SNS-topic-ARN> --protocol email --notification-endpoint <your-email@example.com>
+   ```
+
+By following these steps, you can effectively monitor and prevent any expired domains in Route 53 using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent expired domains in AWS Route 53 using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK (Boto3) and Authentication:**
+   - Install the Boto3 library if you haven't already.
+   - Configure your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+   from botocore.exceptions import NoCredentialsError, PartialCredentialsError
+
+   # Initialize a session using Amazon Route 53
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>